choccy_chip/src/emulator/rom.rs

```diff
@@ -0,0 +1,40 @@
+use std::{fs::File, io::Read, path::PathBuf};
+struct RomParser {
+    file_path: PathBuf,
+}
+
+impl RomParser {
+    /// Creates a new instance of the RomParser struct.
+    ///
+    /// # Arguments
+    ///
+    /// * `file_path` - The path to the ROM file.
+    ///
+    /// # Returns
+    ///
+    /// A new instance of the RomParser struct.
+    pub fn new(file_path: PathBuf) -> Self {
+        Self { file_path }
+    }
+
+    /// Reads the ROM file and returns a vector of bytes.
+    ///
+    /// # Returns
+    ///
+    /// A vector of bytes representing the ROM file.
+    fn read(&self) -> Result<Vec<u8>, &str> {
```

Comment:
I think for the sake of type parameters, you should Make a Rom Struct that is just an Alias to Vec<u8>

Reply:
True.